### DISCUSSION TITLE
Query builder to run alongside Prisma

### DISCUSSION THREAD

**Discussion by KarolScibior** [prisma/prisma#18583] [Q&A] [answered]
Hey, 

I come from MikroORM and I'm loving Prisma so far, but lack of any query builder and no plans to create one really makes me think about going back.

Has anyone used any additional type-safe query builder libraries alongside Prisma to somehow tackle this issue? Like a query builder that would take a type generated by Prisma Client and would somehow let me create a query to run on that DB. 

I'm gonna try [kysely](https://github.com/kysely-org/kysely) with [prisma-kysely](https://www.npmjs.com/package/prisma-kysely), but maybe someone has a better alternatives?

**Comment by nurul3101** [comment, marked as answer]
Hello @KarolScibior 👋 

We do have a feature request for adding support for a native query builder:
- https://github.com/prisma/prisma/issues/5052

If you can add a 👍 to the feature request, then it would help our product team in prioritising it.

I have seen users using [Knex](https://knexjs.org/) in conjunction with Prisma, but I believe kysely would be a good alternative as it has better typescript support.

**Reply by nikolasburk** [reply]
To add to Nurul's answer: 

This tool isn't really a query builder but it still substantially improves the workflow when working with raw queries in Prisma: https://safeql.dev/compatibility/prisma.html We just had the author on [What's new in Prisma](https://www.youtube.com/live/Ls4qDhRuyko?feature=share&t=489) yesterday in case you're interested in learning some more about the tool 😄 

Otherwise, we have some additional open issues for this functionality where it would be helpful if you could leave a 👍 and a comment with your use case to help us prioritize these in the future 🙂 

- https://github.com/prisma/prisma/issues/5560
- https://github.com/prisma/prisma/issues/12279
- https://github.com/prisma/prisma/issues/6689
- https://github.com/prisma/prisma/issues/5761

**Comment by danil-iglu** [comment]
@KarolScibior will this do the trick? 
[https://www.npmjs.com/package/prisma-query-builder-ui](https://www.npmjs.com/package/prisma-query-builder-ui)
